hazelcast/src/test/java/com/hazelcast/test/jdbc/GenericMapStoreObjectProvider.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.test.jdbc;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Allows creating of objects (tables, collections) used in GenericMapStore tests.
+ */
+public interface GenericMapStoreObjectProvider extends TestDatabaseProvider {
```

Comment:
I like the idea that we would have something that is able to create and verify objects in the native database, but I am not sure if we should tie it with the `DatabaseProvider`. Maybe the GenericMapStoreObjectProvider can use the database provider, but it should be a separate object.

Reply:
Done :) Multiple revisions, so just please take a look overall